BinaryEncoding.md

```diff
@@ -65,20 +65,40 @@ Yes:
 
 ## Global structure
 
-* A module contains:
-  * a header followed by
-  * a table (sorted by offset) containing, for each section, its type and offset (within the module), followed by
-  * a sequence of sections.
+* A module contains (in this order):
+  - A header, containing:
+    + The [magic number](https://en.wikipedia.org/wiki/Magic_number_%28programming%29)
+    + Other data TBD (possibly entrypoint, memory bitness, source language, etc.)
+  - A table (sorted by offset) containing, for each section:
+    + A string literal section type name
+    + 32-bit offset within the module
```

Comment:
@titzer That sounds extremely painful. I'd rather not put someone else through the same, so ok sure, I'll make code sections 64 bit as well.